test/utils_test.go

```diff
@@ -328,34 +329,41 @@ func sendTxn(workerID int, txnNum int, routers []*router.Router) {
 	}
 }
 
-func PullFromAssemblers(t *testing.T, userConfig *armageddon.UserConfig, parties []types.PartyID, startBlock uint64, endBlock uint64, transactions int, blocks int, errString string) {
+func PullFromAssemblers(t *testing.T, userConfig *armageddon.UserConfig, parties []types.PartyID, startBlock uint64, endBlock uint64, transactions int, blocks int, errString string) map[types.PartyID][]types.BatchID {
 	var waitForPullDone sync.WaitGroup
+	blockInfos := make(map[types.PartyID][]types.BatchID, len(parties))
 
 	for _, partyID := range parties {
 		waitForPullDone.Add(1)
 
 		go func() {
 			defer waitForPullDone.Done()
 
-			totalTxs, totalBlocks, err := PullFromAssembler(t, userConfig, partyID, startBlock, endBlock, transactions, blocks)
+			totalTxs, totalBlocks, bInfos, err := PullFromAssembler(t, userConfig, partyID, startBlock, endBlock, transactions, blocks)
+			blockInfos[partyID] = bInfos
 			errString := fmt.Sprintf(errString, partyID)
 			require.ErrorContains(t, err, errString)
-			require.Equal(t, uint64(transactions), totalTxs)
-			require.Equal(t, uint64(blocks), totalBlocks)
+			// TODO:  check that we get all of the submitted transactions
+			require.LessOrEqual(t, uint64(transactions), totalTxs)
+			if blocks > 0 {
+				require.LessOrEqual(t, uint64(blocks), totalBlocks)
+			}
 		}()
 	}
 
 	waitForPullDone.Wait()
+	return blockInfos
 }
 
-func PullFromAssembler(t *testing.T, userConfig *armageddon.UserConfig, partyID types.PartyID, startBlock uint64, endBlock uint64, transactions int, blocks int) (uint64, uint64, error) {
+func PullFromAssembler(t *testing.T, userConfig *armageddon.UserConfig, partyID types.PartyID, startBlock uint64, endBlock uint64, transactions int, blocks int) (uint64, uint64, []types.BatchID, error) {
 	require.NotNil(t, userConfig)
 	dc := client.NewDeliverClient(userConfig)
-	toCtx, toCancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
I thought we are going to use a variable to set the timeout 
Otherwise waiting every time a minute may make the tests too long